compiler/hsSyn/HsTypes.hs

```diff
@@ -1097,6 +1097,8 @@ mkHsAppTys = foldl' mkHsAppTy
 --      splitHsFunType (a -> (b -> c)) = ([a,b], c)
 -- Also deals with (->) t1 t2; that is why it only works on LHsType Name
 --   (see Trac #9096)
+--
+-- TODO Krzysztof: we might allow FUN for consistency.
```

Comment:
TODO needs to disappear from the final merge. Turn it into an issue before submitting.